sys/shell/commands/sc_netif.c

```diff
@@ -742,9 +748,15 @@ static int _netif_add(char *cmd_name, kernel_pid_t dev, int argc, char **argv)
         return 1;
     }
 
-    if (gnrc_ipv6_netif_add_addr(dev, &addr, prefix_len, (type == _ANYCAST) ?
-                                 GNRC_IPV6_NETIF_ADDR_FLAGS_NON_UNICAST :
-                                 GNRC_IPV6_NETIF_ADDR_FLAGS_UNICAST) == NULL) {
+    flags = GNRC_IPV6_NETIF_ADDR_FLAGS_NDP_AUTO;
+    flags |= ((type == _ANYCAST) ? GNRC_IPV6_NETIF_ADDR_FLAGS_NON_UNICAST :
+                                   GNRC_IPV6_NETIF_ADDR_FLAGS_UNICAST);
+
+    if (!(iface->flags & GNRC_IPV6_NETIF_FLAGS_SIXLOWPAN)) {
+        flags |= GNRC_IPV6_NETIF_ADDR_FLAGS_NDP_ON_LINK;
```

Comment:
This should not be a property of the address, but determined on sending, as it was before (just with the interface check).